### PR TITLE
Switch to kotlin-stdlib-jdk8, update gradle, AGP, kotlin, fix warnings

### PR DESCRIPTION
**Pull request by edenman**

### CODE REVIEW

**Comment by edenman** [comment]
Addresses #547 (thanks, @jaredsburrows!)

**Comment by jaredsburrows** [comment]
@edenman Thank you. The sad part is in projects that use AGP 3.2.0, people will have this problem again:
```
$ gradlew dependencies | grep -is jre
<-------------> 0% EXECUTING [2s]
> :dependencies > Resolve dependencies of :compileClasspath
     |    |         |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0
     |    |         |         \--- org.jetbrains.kotlin:kotlin-stdlib-jre7:1.2.0
     |    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    |         |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0
     |    |         |         \--- org.jetbrains.kotlin:kotlin-stdlib-jre7:1.2.0
     |    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
     |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
<-------------> 0% EXECUTING [2s]in:kotlin-stdlib-jre8:1.2.0 (*)
> :dependencies > Resolve dependencies of :default
|    |    |         |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0
|    |    |         |         \--- org.jetbrains.kotlin:kotlin-stdlib-jre7:1.2.0
|    |    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    |         |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0
|    |    |         |         \--- org.jetbrains.kotlin:kotlin-stdlib-jre7:1.2.0
|    |    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    |    \--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
|    |    +--- org.jetbrains.kotlin:kotlin-stdlib-jre8:1.2.0 (*)
```

Filed bug here: https://issuetracker.google.com/issues/117136845.

File: build.gradle [hunk 1]
```diff
@@ -45,7 +44,7 @@ buildscript {
   ]
 
   dependencies {
-    classpath 'com.android.tools.build:gradle:3.0.1'
+    classpath 'com.android.tools.build:gradle:3.3.0-alpha12'
```
Comment (JakeWharton) [inline build.gradle]: Do we need this? Why not 3.2.0?

Comment (edenman) [reply build.gradle]: Don't feel strongly, I just do what AS tells me to

Comment (jaredsburrows) [reply build.gradle]: @edenman @JakeWharton To fix the warning, it is currently fixed in `3.3.0-alpha13`.

**Pull request merged**